packages/generator/tests/acceptance/test_streaming_measurement_client.py

```diff
@@ -49,6 +51,64 @@ def test___measurement_plugin_client___stream_measure___returns_output(
     assert responses == expected_output
 
 
+def test___measurement_plugin_client___invoke_measure_from_two_threads___initiates_first_measure_and_rejects_second_measure(
+    measurement_plugin_client_module: ModuleType,
+) -> None:
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+
+    with pytest.raises(RuntimeError) as exc_info:
+        measure_thread = threading.Thread(target=measurement_plugin_client.measure)
+        measure_thread.start()
+        measurement_plugin_client.measure()  # Calling Measure in main thread, to catch error in pytest.
```

Comment:
This is a race condition. There is no guarantee that the background thread calls measure first.

Actually it's two race conditions, if the call to measure() takes a finite amount of time.

You might be better off using concurrent.futures.ThreadPoolExecutor to make two concurrent calls to measure, then assert that one of them raises the exception.